src/main/java/com/algorand/sdkutils/utils/TypeDef.java

```diff
@@ -49,7 +47,50 @@ public TypeDef(
         this.openApiFormat = openApiFormat;
         this.openApiAlgorandFormat = openApiAlgorandFormat;
         this.openApiAlgorandGoFormat = openApiAlgorandGoFormat;
-        this.enumValues = null;
+        this.required = required;
+    }    
+    /**
+     * @param javaTypeName is the generated code type: e.g. List<abc>, or MyEnumClassName
+     * @param rawTypeName is the type name from the spec file
+     * @param type is a loosely defined tag used by generator e.g. enum, array, etc.
+     * @param propertyName is the class/struct member name
+     * @param goPropertyName, is when provided with x-go-name
+     * @param doc is the comments associated with the parameter
+     * @param required indicates if the field is a required field
+     * @param enumValues is passed if the type is an array of enums
+     */
+    public TypeDef(
```

Comment:
Could you add one extra field to the other constructor instead of making a copy?